### DISCUSSION TITLE
how to "link" two datasets?

### DISCUSSION THREAD

**Discussion by ergonyc** [ndcn/omicser#185] [General]
The group discussion (Jan, Brain Trust Meeting) let do a consensus to refocus on extending the functionality of the browser tool IMMEDIATELY with the __Ability to link across datasets__.

How are we going to do this?   

Caveat:  we will look to "link" datasets only... though we should anticipate that on the roadmap (v3.0) we will want to harmonize and link / integrate _across_  -omic domains.

- What is the linking proposition?
Q1: create an NDCN "linking table"/algorithm vs. a preferred API?
    - potential APIs?
    - omic-specific tables?

Q2: Potential Experts on "linking" or -omic ontologies to leverage?

**Comment by ergonyc** [comment]
@ddifraia @ricoderks @bnovotny any thoughts?

**Comment by ergonyc** [comment]
@MenonLab shared this perspective via e-mail:

<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-36ef70f2-7fff-f328-e0df-e4577ea6f089">

Modality 1 | Modality 2 | Basic | Advanced
-- | -- | -- | --
Proteomics | Transcriptomics | Gene-to-protein name mapping | Isoform-level mapping of genes and proteins
Proteomics/Transcriptomics | Lipidomics | Genes/proteins known to be involved in synthesis of individual lipids – mapped to each lipid | Proteins involved in general lipid regulation pathways – lipids involved in same pathways, aggregated by pathway
Proteomics/Transcriptomics | Metabolomics | Proteins known to be involved in synthesis/regulation of individual metabolites – mapped to each metabolite | Proteins involved in general metabolic pathways – metabolites involved in same pathways, aggregated by pathway
Lipidomics | Metabolomics | Lipids mapped to metabolic products and vice versa (non-unique mapping) | Lipids and metabolites involved in same general pathways, aggregated for each pathway

</b>

**Reply by ergonyc** [reply]
@MenonLab do you have a preferred API or join tables to execute these mappings?

**Reply by MenonLab** [reply]
@ergonyc only for the first row - we can use the Refseq gene-protein mapping tables there. For the other rows, it's more speculative, and would require some biological insight to define the mappings

**Reply by ddifraia** [reply]
@ergonyc , for mapping purpose in R I usually use AnnotationHub https://bioconductor.org/packages/release/bioc/html/AnnotationHub.html

Is very deep and has a lot of different interesting annotation, besides ID mapping, can be used to retrieve information about, GO terms, pathways, and disease relevant genes between others. 

Can be easily used to map transcript/genes to protein/UNIPROT entries. 
In this case one can think of storing the saved tables, and access them for mapping. One could ideally also think of automatically detecting the feature types, by checking the which types of the ID in the annotation table gives the most hit.

**Reply by ergonyc** [reply]
@ddifraia @ricoderks Lets go ahead and pull in this into the code to drive the linking. 

- What are the other browser internals that we need to "persist" between datasets to properly make a linking? 
- what kind of interface to we need to build for the annotations/mapping?

**Reply by ricoderks** [reply]
For linking lipids to proteins this database from lipidmaps is maybe very interesting: https://lipidmaps.org/databases/lmpd/overview

Some other maybe interesting packages:

* An R package to translate between several id's is [webchem](https://lipidmaps.org/databases/lmpd/overview).
* To search [kegg](http://www.bioconductor.org/packages/release/bioc/html/KEGGREST.html). 
* a general interface for [databases](https://github.com/pkrog/biodb/)

**Reply by ricoderks** [reply]
I think during the curation we already need some form of general ID number or the proteins/genes/lipids/metabolites so that we can use them.

**Reply by ddifraia** [reply]
@ergonyc I think the best way in this case would be to download species specific tables. For human, mouse, and at least some of the key model organism. Without having to connect to the database everytime. In this case the user would need to specify the species of each experiment thou...

**Reply by MenonLab** [reply]
Lipidmaps looks very interesting indeed, and would be a nice way to map between lipid species and genes/proteins. I've been looking around to see if there is a static table we could download from there

**Reply by ricoderks** [reply]
@MenonLab you can download it. It's a txt file of about 3.5-4Mb. The only thing is, I just saw, that the latest timestap is from 2015! 

@ddifraia @ergonyc I agree, the best thing is to download them and update the tables a few times a year if nescessary.

**Reply by MenonLab** [reply]
@ricoderks - that's great! Even if it hasn't been updated in a few years (2015 seems like a lifetime ago, haha), we can use it as a framework for testing ways to map.

**Comment by bnovotny** [comment]
Hi all, wanted to throw out a resource for linking metabolomics to proteomics and transcriptomics: the Comparative Toxicogenomics Database. Their interactions can be downloaded (https://ctdbase.org/downloads/). It would not be totally comprehensive but might be a good supplementary resource.

**Reply by ergonyc** [reply]
Cool resource.  Do you think that we would want to have a master table to join against, or query the online DB on the fly?

**Reply by ergonyc** [reply]
What other resources are out there that ought to be covered?

**Reply by ricoderks** [reply]
I think the best is, if possible, to download. That avoids causing problems on our side if they change the api or whatever.

**Reply by ergonyc** [reply]
I agree with @ricoderks that for our first iteration of the linking we should start with a fixed and agreed upon table to join against.  Who has two databases to develop against?  Someone needs to code up a first attempt at "linking".

**Reply by ricoderks** [reply]
Is there a data set available?

**Comment by ergonyc** [comment]
Great insight today's "brain trust" discussion about two types of _linking_:

1. categorical / top-down linking 
2. statistical (observed/inferred) linking 

Are we in consensus that we should start with type 1 for now?

**Reply by ricoderks** [reply]
Agree!

**Reply by ddifraia** [reply]
I also agree

**Comment by ergonyc** [comment]
Has anyone tried a proof-of-concept to make such a linking?  
@ricoderks I will send you a private link to some AnswerALS data which has transcriptomics AND proteomics.

@ddifraia  @mtiessen1175  I'm assuming that you have some available proteomic/transcriptomic data sets you can test with?

@bnovotny ?

**Reply by bnovotny** [reply]
Hi, I've found a more comprehensive file for linking metabolites and proteins (from [STITCH](http://stitch.embl.de/)) so I can try a test of linking some of our metabolomic and proteomic data. 

Have we pinpointed what "link" means at this point? Once we have a joining table, are we joining metadata? DE results? Count/abundance data? All of the above? Trying to visualize the final product of this "version 1" linking to determine how to proceed with a proof-of-concept.

**Reply by ricoderks** [reply]
@ergonyc I started yesterday to code something, but need a dataset.

**Reply by ricoderks** [reply]
@bnovotny Good question! I'll talk with Martin and Yassene and see what they think!

**Reply by ddifraia** [reply]
@ergonyc  We have some transcriptome and proteome data, I'll need to ask Alessandro what is the best dataset we have in this regard. Probably some mouse data...
